extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/internal/LifxLightDiscovery.java

```diff
@@ -31,91 +58,408 @@
+                            sendPacket(versionPacket, csp.ipaddress, unicastKey);
+                            LifxNetworkThrottler.unlock();
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("An exception orccurred while communicating with the bulb : '{}'", e.getMessage());
+                e.printStackTrace();
+            }
+        }
+    };
+
+    private void handlePacket(Packet packet, InetSocketAddress address) {
+
+        logger.debug("Discovery : Packet type '{}' received from '{}' for '{}' with sequence '{}' and source '{}'",
+                new Object[] { packet.getClass().getSimpleName(), address.toString(), packet.getTarget().getHex(),
+                        packet.getSequence(), Long.toString(packet.getSource(), 16) });
+
+        if (packet.getSource() == source || packet.getSource() == 0) {
+
+            if (packet instanceof StateServiceResponse) {
+                serviceResponses.put(packet.getTarget(), (StateServiceResponse) packet);
+                int port = (int) ((StateServiceResponse) packet).getPort();
+
+                if (port != 0) {
+                    try {
+                        ConnectionSetupParameter connectionSetup = new ConnectionSetupParameter();
+                        connectionSetup.ipaddress = new InetSocketAddress(address.getAddress(), port);
+                        connectionSetup.target = packet.getTarget();
+                        connectionsToSetUp.add(connectionSetup);
+                    } catch (Exception e) {
+                        logger.warn("An exception occurred while connecting to IP address : '{}'", e.getMessage());
+                        return;
+                    }
+                }
+            }
+
+            if (packet instanceof StateVersionResponse) {
+                StateServiceResponse serviceResponse = serviceResponses.get(packet.getTarget());
+
+                if (serviceResponse != null) {
+                    DiscoveryResult discoveryResult = createDiscoveryResult(serviceResponse,
+                            (StateVersionResponse) packet);
+                    thingDiscovered(discoveryResult);
+                }
+            }
+
+        }
     }
 
-    private DiscoveryResult createDiscoveryResult(LFXLight light) {
-        ThingUID thingUID = getUID(light);
+    private DiscoveryResult createDiscoveryResult(StateServiceResponse packet, StateVersionResponse returnedPacket) {
+
+        MACAddress discoveredAddress = packet.getTarget();
+        ThingUID thingUID = getUID(discoveredAddress.getAsLabel());
+        Products product = Products.getProductFromProductID(returnedPacket.getProduct());
 
-        String label = light.getLabel();
+        String label = "";
 
         if (StringUtils.isBlank(label))
-            label = "LIFX";
+            label = product.getName() + " v." + returnedPacket.getVersion() + " [" + discoveredAddress.getAsLabel()
```

Comment:
This results in a rather ugly label - I would reduce it to `product.getName()`only.
For the address, you can use 'DiscoveryResultBuilder.withRepresentationProperty()' - this makes sure, that the address is also shown to the user as an identification.